### PR TITLE
Added functionality to bootstrap production repos.

### PR DESCRIPTION
**Pull request by Korulag**
This commit brings the ability to bootstrap production repositories in Pulp to be able to release into them later.
Tickets: ALBS:18, ALBS-29

### CODE REVIEW

File: alws/app.py [hunk 1]
```diff
@@ -37,5 +38,5 @@ async def shutdown():
 
 
 for module in (builds, build_node, distro, platforms, projects, releases,
-               tests, users):
+               repositories, tests, users):
```
Comment (soksanichenko) [inline alws/app.py:41]: Why did you not move it to some external constant which will be imported and used here?

File: alws/crud.py [hunk 2]
```diff
@@ -782,3 +788,228 @@ async def commit_release(db: Session, release_id: int) -> (models.Release, str):
         selectinload(models.Release.platform)
     ))
     return release_res.scalars().first(), message
+
+
+async def get_repositories(db: Session, repository_id: int = None
+                           ) -> typing.List[models.Repository]:
+    repo_q = select(models.Repository)
+    if repository_id:
```
Comment (soksanichenko) [inline alws/crud.py:796]: The Zen of Python: Explicit is better than implicit.
So, `repository_id is not None` is better. And it helps to catch errors in the case when repository_id is zero or empty string.

File: alws/crud.py [hunk 3]
```diff
@@ -782,3 +788,228 @@ async def commit_release(db: Session, release_id: int) -> (models.Release, str):
...
+    return repository
+
+
+async def update_repository(
+        db: Session, repository_id: int,
+        payload: repository_schema.RepositoryUpdate
+) -> models.Repository:
+    async with db.begin():
+        result = await db.execute(select(models.Repository).where(
+            models.Repository.id == repository_id))
```
Comment (soksanichenko) [inline alws/crud.py:870]: Why do you not use `get(repository_id`?

**Review by soksanichenko — CHANGES_REQUESTED** [review]

File: alws/crud.py (hunk 3 above)
Comment (Korulag) [reply alws/crud.py:870]: https://stackoverflow.com/a/51997031 in this particular case you're right, but it's not always a correct way. I'll fix it

File: alws/crud.py [hunk 4]
```diff
@@ -782,3 +788,235 @@ async def commit_release(db: Session, release_id: int) -> (models.Release, str):
...
+
+
+async def update_repository(
+        db: Session, repository_id: int,
+        payload: repository_schema.RepositoryUpdate
+) -> models.Repository:
+    async with db.begin():
+        repository = await db.execute(select(
+            models.Repository).get(repository_id))
+        # repository = result.scalars().first()
```
Comment (anfimovdm) [inline alws/crud.py:889]: Is that mistake?

**Review by anfimovdm — APPROVED** [review]

**Review by anfimovdm — APPROVED** [review]

File: alws/crud.py [hunk 5]
```diff
@@ -867,3 +874,223 @@ async def commit_release(db: Session, release_id: int) -> (models.Release, str):
...
+            models.Repository.id == repository_id))
+        await db.commit()
+
+
+async def add_to_platform(db: Session, platform_id: int,
+                          repository_ids: typing.List[int]) -> models.Platform:
+    platform_result = await db.execute(select(models.Platform).where(
+        models.Platform.id == platform_id).options(
+        selectinload(models.Platform.repos)).with_for_update())
+    platform = platform_result.scalars().first()
```
Comment (rpisarev) [inline alws/crud.py:995]: Platform could be not found in general case

File: alws/routers/platforms.py [hunk 6]
```diff
@@ -33,3 +33,19 @@ async def modify_platform(
...
+              response_model=platform_schema.Platform)
+async def add_repositories_to_platform(platform_id: int,
+                                       repositories_ids: typing.List[int],
+                                       db: database.Session = Depends(get_db)):
+    return await crud.add_to_platform(db, platform_id, repositories_ids)
+
+
+@router.patch('/{platform_id}/remove-repositories',
+              response_model=platform_schema.Platform)
+async def add_repositories_to_platform(platform_id: int,
```
Comment (rpisarev) [inline alws/routers/platforms.py:48]: It seems you miss renaming the function

Comment (Korulag) [reply alws/routers/platforms.py]: Thanks, fixed

File: alws/crud.py (hunk 5 above)
Comment (Korulag) [reply alws/crud.py:995]: Yeah, you're right, it may be, will fix

**Review by JohnKepplers — APPROVED** [review]

**Review by rpisarev — APPROVED** [review]

**Pull request merged**